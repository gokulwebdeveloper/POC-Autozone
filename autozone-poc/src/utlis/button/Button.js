import React from 'react';
import PropTypes from 'prop-types';

export default function Button({btnText}) {
    return (
        <button className="btn essence-btn">{btnText}</button>
    )
}

Button.propTypes = {
    btnText: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number])
}

Button.defaultProps = {
    text: 'button'
}    