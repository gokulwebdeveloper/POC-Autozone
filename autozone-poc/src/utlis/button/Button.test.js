import React from 'react';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';
//import {unmountComponentAtNode} from 'react-dom';
import Button from './Button';

let btn=null;

//mount component for testing
beforeEach(()=>{
    btn = mount(<Button btnText="button" />);
});

//Cleanup after test execution
afterEach(()=>{
    btn.unmount();
});


describe("test Button component", ()=>{
    //test if component render correctly
    it("component render correctly", ()=>{
        expect(btn.exists()).toBeTruthy();
    });

     //test if it should have one button for button text
     it("should have one button tag", ()=>{
        expect(btn.find('button')).toHaveLength(1);
    });

    //test if it have correct props
    it("should have btnText as props", ()=>{
        expect(btn.prop('btnText')).toEqual('button');
    });

    //SnapShot Match test
    it("SnapShot Match", ()=>{
        const btn = renderer.create(<Button />).toJSON();
        expect(btn).toMatchSnapshot()
    });
});