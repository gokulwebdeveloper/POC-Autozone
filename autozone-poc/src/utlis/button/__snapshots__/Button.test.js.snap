// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Button component SnapShot Match 1`] = `
<button
  className="btn essence-btn"
/>
`;
